owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsLogsFragment.kt

```diff
@@ -0,0 +1,91 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Juan Carlos Garrote Gascón
+ *
+ * Copyright (C) 2021 ownCloud GmbH.
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.presentation.ui.settings.fragments
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.preference.CheckBoxPreference
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.SwitchPreferenceCompat
+import com.owncloud.android.R
+import com.owncloud.android.presentation.viewmodels.settings.SettingsViewModel
+import com.owncloud.android.presentation.ui.settings.LogHistoryActivity
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class SettingsLogsFragment : PreferenceFragmentCompat() {
+
+    // ViewModel
+    private val settingsViewModel by viewModel<SettingsViewModel>()
+
+    private var prefEnableLogging: SwitchPreferenceCompat? = null
+    private var prefHttpLogs: CheckBoxPreference? = null
+    private var prefLogsView: Preference? = null
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.settings_logs, rootKey)
+
+        prefEnableLogging = findPreference(PREFERENCE_ENABLE_LOGGING)
+        prefHttpLogs = findPreference(PREFERENCE_LOG_HTTP)
+        prefLogsView = findPreference(PREFERENCE_LOGGER)
+
+        if (!settingsViewModel.isEnableLoggingOn()) {
+            prefHttpLogs?.isVisible = false
+            prefLogsView?.isVisible = false
+        }
```

Comment:
I chose this name because maybe the one you propose could create confusion since "enabled" is also the name of the state of a UI view (enabled/disabled), but in this case I refer to the internal state of the preference and consequently the state (checked/not checked) of the SwitchPreference. What's your opinion on this?

Reply:
Ok, I see. I think that it should not be a problem at all, since this is a ViewModel call and you know that the UI view is not handled there. But, as you consider. There is no problem for me 👍